app/views/LocationTracking.js

```diff
@@ -36,30 +42,91 @@ class LocationTracking extends Component {
     super(props);
 
     this.state = {
+      timer_intersect: null,
       isLogging: '',
     };
   }
 
   componentDidMount() {
     BackHandler.addEventListener('hardwareBackPress', this.handleBackPress);
     GetStoreData('PARTICIPATE')
-      .then(isParticipating => {
-        console.log(isParticipating);
-
-        if (isParticipating === 'true') {
-          this.setState({
-            isLogging: true,
-          });
-          this.willParticipate();
-        } else {
-          this.setState({
-            isLogging: false,
-          });
-        }
-      })
-      .catch(error => console.log(error));
+        .then(isParticipating => {
+          if (isParticipating === 'true') {
+            this.setState({
+              isLogging: true,
+            });
+            this.willParticipate();
+          } else {
+            this.setState({
+              isLogging: false,
+            });
+          }
+        })
+        .catch(error => console.log(error));
+
+    let timer_intersect = setInterval(this.intersect_tick, 1000 * 60 * 60 * 12); // once every 12 hours
```

Comment:
setting long timeouts and intervals is not recommended. We should use background task aware libs like:  https://www.npmjs.com/package/react-native-background-timer

Reply:
I agree, but this wasn't introduced by the PR.  Not really this contributor's responsibility.

Reply:
this is not changed by me. May be an issue with some space

Reply:
Oops, my bad. I got confused by the branch not being up to date.